python/lsst/jointcal/jointcal.py

```diff
@@ -671,6 +686,17 @@ def _load_reference_catalog(self, refObjLoader, referenceSelector, center, radiu
         else:
             refCat = selected.sourceCat
 
+        if self.config.astrometryReferenceErr is None and 'coord_ra_err' not in refCat.schema:
+            #####################
+            # TODO: is KeyError really the best thing to raise here?
+            # RuntimeError also feels too generic, but may be better?
+            #####################
+            raise KeyError("Reference catalog does not contain coordinate errors, and"
+                           "config.astrometryReferenceErr not supplied.")
```

Comment:
Yeah, I see your Error dilemma. The second half is a KeyError, but 99% of the time this is going to come up when someone forgets to set the `config.astrometryReferenceErr`, so I'd argue the first half is more important. So logically this is a forgotten config parameter that the user needs to supply. I've used `ValueError` for misconfigurations in the past: https://github.com/lsst/pipe_tasks/blob/master/python/lsst/pipe/tasks/assembleCoadd.py#L210

Reply:
I think this probably boils down to making this exception useful on a catch side. If you want your caller/catcher to distinguish between different exception types then it may be better to define special exception class for it. If you are not sure that it will ever be caught then exception type probably does not matter at all, but message itself should be very clear and unambiguous.

Reply:
@yalsayyad : I think the case you cite there in `AssembleCoaddConfig` should be using `pexConfig.FieldValidationError` instead, since it's part of the Config validation. Which, as I think about it, is probably the correct thing to raise here too, since as you say in this case the user probably should have supplied it.

To the "what happens if it's caught" point, I think this is a non-recoverable error that the user will have to sort out, which `FieldValidationError` should make pretty clear.

Reply:
Having done that, that's definitely the right error to raise. The resulting message has a lot of additionally useful information:

```
E           lsst.pex.config.config.FieldValidationError: Field 'astrometryReferenceErr' failed validation: Reference catalog does not contain coordinate errors, and config.astrometryReferenceErr not supplied.
E           For more information read the Field definition at:
E             File jointcal/jointcal.py:299 (JointcalConfig)And the Config definition at:
E             File jointcal/jointcal.py:135 (<module>)
```